internal/handlers/tigris/tjson/double.go

```diff
@@ -50,10 +53,16 @@ func (d *doubleType) UnmarshalJSON(data []byte) error {
 
 // MarshalJSON implements tjsontype interface.
 func (d *doubleType) MarshalJSON() ([]byte, error) {
-	res, err := json.Marshal(float64(*d))
-	if err != nil {
-		return nil, lazyerrors.Error(err)
-	}
+	// This will default to precision of 6
+	//res := []byte(fmt.Sprintf("%f", float64(*d)))
+
+	// get precision of the value to preserve it
+	dStr := strconv.FormatFloat(float64(*d), 'f', -1, 64)
+	precision := len(dStr) - strings.Index(dStr, ".") - 1
+
+	res := []byte(fmt.Sprintf("%."+strconv.Itoa(precision)+"f", float64(*d)))
```

Comment:
```suggestion
	res := []byte(dStr)
```

Reply:
That will force the default precision somewhere I believe

Reply:
Precision is the third argument of `FormatFloat` – `-1` there

Reply:
I didn't remember the exact issue so reproduced it:
```
                                        expected: "179769313486231570814527423731704356798070567525844996598917476803157260780028538760589558632766878171540458953514382464234321326889464182768467546703537516986049910576551282076245490090389328944075868508455133942304583236903222948165808559332123348274797826204144723168738177180919299881250404026184124858368.000000"
                                        actual  : "179769313486231570000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
```
I guess I'll need to understand the function a bit more